src/views/quiz/QuizCreation.vue

```diff
@@ -62,28 +58,33 @@ export default {
   components: {
     InputField,
     InputImage,
-    QuestionCreation,
-    Camera
+    QuestionCreation
   },
   computed: {
     ...mapGetters('Quiz', {
       quiz: 'newQuiz'
     })
   },
   methods: {
-    ...mapActions('Quiz', {
-      create: 'create'
+    ...mapActions({
+      create: 'Quiz/create',
+      upload: 'Image/upload'
     }),
     ...mapMutations('Quiz', {
       addQuestion: ADD_QUESTION,
       setQuiz: SET_QUIZ
     }),
-
     createQuiz () {
       this.create(this.quiz)
       /**
        * @todo Implement error handling for createQuiz here.
+       *
        */
+    },
+    uploadImage (image) {
+      this.upload(image).then((response) => {
+        this.setQuiz(response.data)
```

Comment:
What does that mean setQuiz?

Reply:
I changed names of setQuiz and setQuestion to better represent they actual function. Resolved in
9a5bc6e9e421938c2b0d04d4bb05def35033e22c

Reply:
Actually I reverted this change, because this is mutation which updates quiz with whatever data u throw at it not just with image, in this case it updates only image, but it is also used in this component to set value of quiz like name and description from input fields. This is flexible approach which also reduces code I need to write but the payoff is a little performance because mutation have to check if data has property before assigning it.
Should i use micro mutations like setQuizImage, setQuizDescription and setQuizName instead?

Reply:
Performance is often less important than readability and stability, but not always. SetQuiz and setQuestion methods are like setQuiz, they don't provide specific information about themselves. Consider storing data in vuex store, and making mutations on state.

Reply:
What i were refering to as mutation is this block of code
```suggestion
  [SET_QUIZ] (state, data) {
    if (data.name) {
      state.newQuiz.name = data.name
    }
    if (data.description) {
      state.newQuiz.description = data.description
    }
    if (data.imageUrl) {
      state.newQuiz.imageUrl = data.imageUrl
    }
  },
```

Then i map it here:
```suggestion
    ...mapMutations('Quiz', {
      addQuestion: ADD_QUESTION,
      setQuiz: SET_QUIZ
    }),
```

Should i split mutation from first block of code into 3 micro mutations which set only one property of quiz?
Then I would have to do more mappings ( and more imports for those strings constants):
```suggestion
    ...mapMutations('Quiz', {
      addQuestion: ADD_QUESTION,
      setQuizImage: SET_QUIZ_IMAGE,
      setQuizDescription: SET_QUIZ_DESCRIPTION,
      setQuizName: SET_QUIZ_NAME
    }),
```

but on the other instead of writting 
```suggestion
    uploadImage (image) {
      this.upload(image).then((response) => {
        this.setQuiz(response.data)
      })
    }
```
I could write 
```suggestion
    uploadImage (image) {
      this.upload(image).then((response) => {
        this.setQuizImage(response.data)
      })
    }
```
Which is better because it describes what is happening with response data. But I could just write comment and not have to increase boilerplate.
```suggestion
    uploadImage (image) {
      this.upload(image).then((response) => {
        this.setQuiz(response.data)   // Quiz image is set here. Response contains image url on server.
      })
    }
```